src/transports/tcp.rs

```diff
@@ -311,11 +316,12 @@ impl<Id: PeerId> Transport<Id> for TcpTransport<Id> {
         context: Ctx,
         address: SocketAddr,
         timeout: Duration,
-        _config: &Self::OutConnectionConfig,
+        _config: &Self::TransportConfig,
         message_handler: M,
         handshake_handler: I,
     ) -> PeerNetResult<JoinHandle<PeerNetResult<()>>> {
         let peer_stop_rx = self.peer_stop_rx.clone();
+        let conf = self.config.clone();
```

Comment:
Is it on purpose that you use `self.config` and ignore the config passed as parameters ?
It feels unatural when calling a function with a `config` field that this config is ignored and a struct-tied value is used instead ...

Reply:
good catch